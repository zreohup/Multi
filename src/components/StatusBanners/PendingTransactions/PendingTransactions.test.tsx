import { render, userEvent } from '@/src/tests/test-utils'
import { PendingTransactions } from '.'

describe('PendingTransactions', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('should render the default markup', async () => {
    const user = userEvent.setup()
    const mockedFn = jest.fn()

    const { getByText } = render(<PendingTransactions number={'2'} onPress={mockedFn} />)

    expect(getByText('2')).toBeTruthy()

    await user.press(getByText('Pending transactions'))

    expect(mockedFn).toHaveBeenCalled()
  })

  it('should render the pending transactions in fullWidth layout', () => {
    const container = render(<PendingTransactions number={'2'} fullWidth onPress={jest.fn()} />)

    expect(container).toMatchSnapshot()
  })
})
