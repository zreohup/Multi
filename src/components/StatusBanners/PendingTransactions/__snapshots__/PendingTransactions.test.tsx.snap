// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PendingTransactions should render the pending transactions in fullWidth layout 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="pending-transactions"
  >
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#FFECC2",
            "borderBottomLeftRadius": 5,
            "borderBottomRightRadius": 5,
            "borderTopLeftRadius": 5,
            "borderTopRightRadius": 5,
            "flexDirection": "row",
            "gap": 12,
            "justifyContent": "center",
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
            "width": "100%",
          }
        }
      >
        <View
          testID="alert-start-icon"
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FF8C00",
                "borderBottomLeftRadius": 100000,
                "borderBottomRightRadius": 100000,
                "borderTopLeftRadius": 100000,
                "borderTopRightRadius": 100000,
                "flexDirection": "column",
                "height": 24,
                "justifyContent": "center",
                "maxHeight": 24,
                "maxWidth": 24,
                "minHeight": 24,
                "minWidth": 24,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
                "width": 24,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#6C2D19",
                  "fontFamily": "DMSans-SemiBold",
                  "fontSize": 14,
                }
              }
              suppressHighlighting={true}
            >
              2
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "gap": 4,
            }
          }
        >
          <Text
            style={
              {
                "color": "#6C2D19",
                "fontFamily": "DMSans-SemiBold",
                "fontSize": 14,
              }
            }
            suppressHighlighting={true}
          >
            Pending transactions
          </Text>
        </View>
        <View
          testID="alert-end-icon"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#6C2D19",
                  "fontSize": 20,
                },
                undefined,
                {
                  "fontFamily": "SafeIcons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            î¤¦
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
